library/java/src/io/envoyproxy/envoymobile/engine/types/EnvoyObserver.java

```diff
@@ -45,4 +45,9 @@
    *               execution.
    */
   void onError(EnvoyError error);
+
+  /**
+   * Called when the async HTTP stream has completed.
+   */
+  void onComplete();
```

Comment:
I'm sorta of the opinion that we should drop this for now. We're already surfacing this information via booleans in other callbacks. It has the potential to differ in some unexpected ways from those callbacks if people/interfaces rely on it.

Reply:
This _should_ be our hook the native layer uses to signal to the platform that the native layer is "done" with the observer, right?

I was more or less erroring on the side of mirroring the current `main_interfaces.h`

Reply:
It is our hook for cleaning up, but I don't think it's needed here.